### PR TITLE
Feature/weighted stage scoring

### PR DESCRIPTION
**Pull request by jarekzaj**

### CODE REVIEW

File: src/Elsa.CustomActivities/Activities/Common/RadioModel.cs [hunk 1]
```diff
@@ -1,9 +1,30 @@
...
     public class RadioModel
     {
+        [HeActivityInput(HasCollectedProperties = true,
+            CollectedPropertyType = typeof(RadioTest))]
         public ICollection<RadioRecord> Choices { get; set; } = new List<RadioRecord>();
     }
 
     public record RadioRecord(string Identifier, string Answer, bool IsPrePopulated);
+
+    public class RadioTest
```
[the remark this answers is not in the document]
Comment (jarekzaj) [reply src/Elsa.CustomActivities/Activities/Common/RadioModel.cs:17]: renamed

File: src/Elsa.CustomActivities/Activities/Common/WeightedCheckboxModel.cs [hunk 2]
```diff
@@ -8,8 +8,17 @@ namespace Elsa.CustomActivities.Activities.Common
...
+    {
+        public string GroupIdentifier { get; set; } = null!;
         public ICollection<WeightedCheckboxRecord> Choices { get; set; } = new List<WeightedCheckboxRecord>();
+        public decimal? MaxGroupScore { get; set; }
+        public List<decimal>? GroupArrayScore { get; set; }
+
     }
 
-    public record WeightedCheckboxRecord(string Identifier, string Answer, bool IsSingle, string Score, bool IsPrePopulated);
+    public record WeightedCheckboxRecord(string Identifier, string Answer, bool IsSingle, decimal Score, bool IsPrePopulated);
```
Comment (Jonnythan-t) [inline src/Elsa.CustomActivities/Activities/Common/WeightedCheckboxModel.cs:23]: Should this be a string?

Comment (jarekzaj) [reply src/Elsa.CustomActivities/Activities/Common/WeightedCheckboxModel.cs:23]: no, I don't think so, it should be decimal

File: src/Elsa.CustomActivities/Activities/QuestionScreen/QuestionScreen.cs [hunk 3]
```diff
@@ -66,6 +76,49 @@ protected override async ValueTask<IActivityExecutionResult> OnResumeAsync(Activ
                 new SuspendResult()
             }));
         }
+
+        private async Task UpdateQuestionScores(ActivityExecutionContext context)
```
[the remark this answers is not in the document]
Comment (jarekzaj) [reply src/Elsa.CustomActivities/Activities/QuestionScreen/QuestionScreen.cs:80]: no, this calculates the score for each individual question and saves it in the database

File: src/Elsa.CustomActivities/Activities/QuestionScreen/Helpers/CheckboxQuestionHelper.cs [hunk 4]
```diff
@@ -33,7 +33,7 @@ public async Task<string> GetAnswer(string correlationId, string workflowName, s
                         correlationId, questionId, CancellationToken.None);
 
                     if (question != null && question.Answers != null &&
-                        question.QuestionType == QuestionTypeConstants.CheckboxQuestion)
+                        (question.QuestionType == QuestionTypeConstants.CheckboxQuestion || question.QuestionType == QuestionTypeConstants.WeightedCheckboxQuestion))
```
[the remark this answers is not in the document]
Comment (jarekzaj) [reply src/Elsa.CustomActivities/Activities/QuestionScreen/Helpers/CheckboxQuestionHelper.cs:36]: will revisit after merge

[the remark this answers is not in the document]
Comment (jarekzaj) [reply src/Elsa.CustomActivities/Activities/QuestionScreen/Helpers/CheckboxQuestionHelper.cs:36]: done

**Review by Jonnythan-t — APPROVED** [review]

**Pull request merged**